### PR TITLE
JAMES-3692 Write a security checklist

### PR DESCRIPTION
**Pull request by chibenwa**
This document will present the most threats running James exposes you to, and then will list actions one can take to address those threats. We will rate those from MUST (if not addressed basic security features will not be achieved), SHOULD (easy to set up, not strictly required but still very beneficial) and COULD (for the most stringent environments, but expect advanced configuration to be mobilized...)

This document should enable one to see what to expect for James security-wise.

### CODE REVIEW

File: server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ - 4. We recommend setting up anti-spam and anti-virus solutions. James comes with some xref:configure/spam.adoc[SpamAssassin]
+integration, and some experimental, insufficiently tested xref:configure/mailets.adoc#_clamavscan[ClamAV] tooling exisits.
+Filtering with third party systems upstream is also possible.
+
+ - 5. In order to limit your attack surface, disable protocols you or your users do not use. This includes the JMAP protocol,
+POP3, ManagedSieve, etc... Be conservative on what you expose.
+
+=== Best practice: Could
+
+ - 1. Set up link:https://openid.net/connect/[OIDC] for IMAP, SMTP and JMAP. Disable login/plain/basic authentication.
```
Comment (Arsnael) [inline server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc:89]: That I'm not sure it's a good practice, it depends more about the platform requirements... I would not see the point of putting oidc behing a standalone james server for example

Comment (chibenwa) [reply server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc:89]: It is a "could". If you have stringent security requirements, you "could" do it.

**Review by Arsnael — APPROVED** [review]

File: server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+ - 2. You can configure xref:configure/ssl.adoc#_client_authentication_via_certificates[Client authentication via certificates].
```
Comment (vttranlina) [inline server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc:91]: I can't access to href link from Github. (Page not found)
It redirects to `https://github.com/apache/james-project/blob/900c2181b650653e49eea3ef3ff0a0bb61668b51/server/apps/distributed-app/docs/modules/ROOT/pages/operate/configure/ssl.adoc#_client_authentication_via_certificates` 

The correct path should be `pages/operate/configure/ssl.adoc` -> `pages/configure/ssl.adoc`

File: server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+ - 3. You can xref:configure/mailets.adoc#_smimesign[sign], xref:configure/mailets.adoc#_smimechecksignature[verify]
```
Comment (vttranlina) [inline server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc:93]: same
I can't access to href link from Github. (Page not found)

File: server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+Operating an email server exposes you to the following threats:
+
+ - Spammers might attempt to use your servers to send their spam messages on their behalf. We speak of
+*open relay*. In addition to the resources consumed being an open relay will affect the trust other mail
+installations have in you, and thus will cause legitimate traffic to be rejected.
+ - Emails mostly consist of private data, which shall only be accessed by their legitimate user. Failure
+to do so might result in *information disclosure*.
+ - *Email forgery*. An attacker might craft an email on the behalf of legitimate users.
+ - Email protocols allow user to authenticate and thus can be used as *oracles* to guess user passwords.
+ - *Spam*. Non legitimate traffic can be a real bourbon to your users.
```
Comment (ottoka) [inline server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc:17]: Your spellchecker seems to like fine spirits. I believe this should be "burden" instead.

Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc:17]:
```suggestion
 - *Spam*. Non legitimate traffic can be a real burden to your users.
```

**Review by vttranlina — APPROVED** [review]

**Comment by chibenwa** [comment]
If there is not more remarks regarding this, can I merge it?

**Review by quantranhong1999 — APPROVED** [review]

**Comment by chibenwa** [comment]
If there remains comments, I would be happy to answer them in a dedicated PR ;-)

**Pull request merged**